packages/lodestar/src/sync/regular/oneRangeAhead/oneRangeAhead.ts

```diff
@@ -0,0 +1,156 @@
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {SignedBeaconBlock, Slot} from "@chainsafe/lodestar-types";
+import {AbortController, AbortSignal} from "abort-controller";
+import {ILogger, sleep} from "@chainsafe/lodestar-utils";
+import {toHexString} from "@chainsafe/ssz";
+import {EventEmitter} from "events";
+import PeerId from "peer-id";
+import {IRegularSync, IRegularSyncOptions, RegularSyncEventEmitter} from "..";
+import {ChainEvent, IBeaconChain} from "../../../chain";
+import {INetwork} from "../../../network";
+import {GossipEvent} from "../../../network/gossip/constants";
+import {checkBestPeer, getBestPeer} from "../../utils";
+import {getSyncPeers} from "../../utils/peers";
+import {BlockRangeFetcher} from "./fetcher";
+import {BlockRangeProcessor} from "./processor";
+import {ISyncCheckpoint} from "../../interface";
+import {IBlockRangeFetcher, IBlockRangeProcessor, ORARegularSyncModules} from "./interface";
+
+/**
+ * One Range Ahead regular sync: fetch one range in advance and buffer blocks.
+ * Fetch next range and process blocks at the same time.
+ */
+export class ORARegularSync extends (EventEmitter as {new (): RegularSyncEventEmitter}) implements IRegularSync {
+  private readonly config: IBeaconConfig;
+
+  private readonly network: INetwork;
+
+  private readonly chain: IBeaconChain;
+
+  private readonly logger: ILogger;
+
+  private bestPeer: PeerId | undefined;
+
+  private fetcher: IBlockRangeFetcher;
+
+  private processor: IBlockRangeProcessor;
+
+  private controller!: AbortController;
+
+  private blockBuffer: SignedBeaconBlock[];
```

Comment:
For consistency with other Lodestar classes let's condense the property declarations without newlines